### ISSUE TITLE
Duplicate date markers appear for 1 frame on opening room.

### ISSUE THREAD

**Issue by ara4n** [element-hq/element-x-ios#2701]
### Steps to reproduce

1. Open a room (after some kind of sync discontinuity - "it's a cold cache + sliding sync + ss pagination increase + backpagination magic" thing according to @stefanceriu)
2. Observe that for a single frame just as the room slides into place, the date markers appear in duplicate. The bubbles don't have a chance to animate their position before they disappear
3. Ends up feeling flickery and unpolished.

![Screenshot 2024-04-16 at 16 01 22](https://github.com/element-hq/element-x-ios/assets/1294269/d025dbc6-4bc6-48d4-8022-f70e4e497ed3)

I've enabled `matrix_sdk_ui::timeline::day_dividers=trace` log level and will rageshake if/when i next see it.

#### What did you expect?

No rogue duplicate date markers and unpolished weird flickering timeline glitches

### Application version

574

### Will you send logs?

Yes

**Comment by ara4n** [comment]
(a naughty workaround presumably would be to fade in the date markers to cheekily camouflage the bug)

**Comment by pixlwave** [comment]
I think I may have come across something related to this as part of #2771: I could reliably make the table view show stale items by updating the timeline as follows:

```
DispatchQueue.main.sync {
    timelineItems.removeAll() // Removed items would be shown in the table view, all in the wrong place.
}

DispatchQueue.main.sync {
    timelineItems = updatedTimelineItems
}
```

This was reproducible at both the `RoomTimelineController` and `RoomScreenViewModel` levels, but possibly had something to do with the number of items that were being handled.

TL;DR, the timeline table view doesn't like being cleared, and I would presume the situation above is caused by the SDK sending a `reset` diff after opening the room with existing items.

**Comment by manuroe** [comment]
It may be a UIKit issue. The best option would be switch to a swiftUI timeline version 7.
